posts/2024-xx-xx-spring-boot-3.adoc

```diff
@@ -0,0 +1,223 @@
+---
+layout: post
+title: "Running a Spring Boot 3.x application WAR file on Liberty"
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/scottkurz
+author_github: https://github.com/scottkurz
+seo-title: Running a Spring Boot 3.x application WAR file on Liberty
+seo-description: TODO In version 23.0.0.9 and later, you can easily run a Spring Boot 3.x application from a WAR file on Open Liberty. This post use the Spring Boot Petclinic sample application and the Liberty Maven plugin to create and run a WAR-based Spring Boot 3.1 application on Open Liberty. 
+blog_description: TODO In version 23.0.0.9 and later, you can easily run a Spring Boot 3.x application from a WAR file on Open Liberty. This post uses the Spring Boot Petclinic sample application and the Liberty Maven plugin to create and run a WAR-based Spring Boot 3.1 application on Open Liberty. 
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+open-graph-image-alt: Open Liberty Logo
+---
+= Running a Spring Boot 3.x application WAR file on Liberty
+Scott Kurz <https://github.com/scottkurz>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+//Blank line here is necessary before starting the body of the post.
+
+As of Liberty version 23.0.0.9 there are now two ways to deploy Spring Boot WAR files on Liberty.
+
+While a Spring Boot WAR can be deployed like any standard Jakarta EE WAR, it can also be deployed using an optimized deployment for Spring Boot applications. This deployment adds some additional configuration options, and provides advantages for efficiently building Docker container images.
```

Comment:
```suggestion
While a Spring Boot WAR can be deployed like any standard Jakarta EE WAR, it can also be deployed using an optimized deployment for Spring Boot applications. The optimized deployment adds some additional configuration options, and provides advantages for efficiently building Docker container images.
```

Reply:
Also, why would you ever use a non-optimized approach? It sounds, um, sub-optimal...

Reply:
* This part "..the WAR can be optimized deployment .." read a bit awkward so I took your suggestion and tweaked a bit


"Also, why would you ever use a non-optimized approach? It sounds, um, sub-optimal..."

Because:
 * you need extra pom.xml config
 * you need special server.xml config
 * you can't use dev mode

and because you might want to deploy like any WAR.


I added:
"using some extra pom.xml and server.xml configuration."

to maybe thread the needle here or feel free to suggest another wording.